Modules/pyexpat.c

```diff
@@ -1881,9 +1953,21 @@ MODULE_INITFUNC(void)
 
     /* export using capsule */
     capi_object = PyCapsule_New(&capi, PyExpat_CAPSULE_NAME, NULL);
-    if (capi_object)
-        PyModule_AddObject(m, "expat_CAPI", capi_object);
-    return m;
+    if (capi_object) {
+        if (PyModule_AddObject(mod, "expat_CAPI", capi_object) < 0) {
+            goto error;
+        }
+    }
+    return 0;
+
+error:
+    Py_CLEAR(errmod_name);
```

Comment:
On success (return 0), errmod_name leaks 1 reference.

I suggest a pattern like:

```
  int res;

  (...)

  res = 0;
  goto done;

error:
  res = -1;

done:
  Py_CLEAR ...
  return res;
```

Reply:
@vstinner you're right.  I think I'll go back to  your earlier recommendation about doing this closer to where it is used, it actually seems cleaner this way.